### DISCUSSION TITLE
LFX Mentorship 2022 03-Sept-Nov Challenge - For Node API support for WasmEdge QuickJS

### DISCUSSION THREAD

**Discussion by alabulei1** [WasmEdge/WasmEdge#1753] [Show and tell]
For those who want to apply for LFX mentorship for #1745, this is a pretest for the application.
This issue is to add Node API support for WasmEdge. So As a "test",  please implement a nodejs api in WasmEdge QuickJS using our Rust API. You could choose any of the unimplemented nodejs API [here](https://github.com/WasmEdge/WasmEdge/issues/1535).

### Requirements

1. fork [wasmedge-quickjs repository](https://github.com/second-state/wasmedge-quickjs). You should implement a Node API in the repo.
2. Post your repo's GitHub link as a comment under this discussion.
 
### Deadline

Please refer to [the timeline of LFX mentorship 2022 03-Sept-Nov](https://github.com/cncf/mentoring/tree/main/lfx-mentorship/2022/03-Sept-Nov#timeline).
The final date of valid submissions will be August 31, 5:00 PM PDT

### Reference

https://github.com/second-state/wasmedge-quickjs

**Comment by sandeepB3** [comment]
Hello there, I would be interested to be a part of the LFX mentorship under this project. I shall learn and implement a Node API then make a PR soon. Is there any other criteria to be fulfilled for the same ?

**Comment by Puellaquae** [comment]
Hi @alabulei1 . I want to take a part in this project. It's ok to use this  https://github.com/second-state/wasmedge-quickjs/pull/64 as my challenge or should I implement another one?

**Reply by alabulei1** [reply]
> Hi @alabulei1 . I want to take a part in this project. It's ok to use this [second-state/wasmedge-quickjs#64](https://github.com/second-state/wasmedge-quickjs/pull/64) as my challenge or should I implement another one?

Hi  @Puellaquae, I think [second-state/wasmedge-quickjs#64](https://github.com/second-state/wasmedge-quickjs/pull/64) is OK.

**Comment by Aviii06** [comment]
HI @alabulei1. Here is the PR link: **https://github.com/second-state/wasmedge-quickjs/pull/82**
